app/src/main/java/com/example/taskyy/domain/navigation/Navigation.kt

```diff
@@ -26,24 +29,53 @@ fun Nav() {
             composable(route = Screen.Login.route) {
                 val loginViewModel = hiltViewModel<LoginViewModel>()
                 val state = loginViewModel.state
-                LoginScreen(state, loginViewModel::onEvent, navController)
+                LoginScreen(
+                    state = state,
+                    onEvent = { event ->
+                        when (event) {
+                            is LoginEvent.OnLoginClick -> if (state.isLoginSuccessful) {
+                                (navController.navigate(Screen.Agenda.route))
+                            }
```

Comment:
For logging in it's a bit different. Here you only want to navigate when logging was successful and not generally after clicking the button. When you click the button for the first time here, the login wasn't successful yet, so nothing will happen.
Instead, you need to work with events the ViewModel sends to the UI. You can do that with a Channel:
```
class LoginViewModel: ViewModel() {

    private val eventChannel = Channel<LoginEvent>() // Different set of events
    val events = eventChannel.receiveAsFlow()
    
    fun login() {
        // Do login...
        
        if(loginSuccess) {
            eventChannel.send(LoginEvent.LoginSuccess)
        }
    }
}
```
Then you can use this helper composable in your composable block in the nav host to listen to these events and do the navigation there:
```
@Composable
private fun <T> ObserveAsEvents(flow: Flow<T>, onEvent: (T) -> Unit) {
    val lifecycleOwner = LocalLifecycleOwner.current
    LaunchedEffect(flow, lifecycleOwner.lifecycle) {
        lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
            withContext(Dispatchers.Main.immediate) {
                flow.collect(onEvent)
            }
        }
    }
}
```
In your Navhost:
```
composable("login") {
    val viewModel = hiltViewModel<LoginViewModel>()
    ObserveAsEvents(viewModel.events) { event ->
        when(event) {
            is LoginEvent.Login -> // Navigate here
        }
    }
}
```
So to summarize:
Instant navigation after button click? -> You can navigate in the UI directly
Navigation only after an asynchronous call in the VM? -> Navigate with via Channel as above